### PR TITLE
[ticket/13911] Add events to configure options for profile fields

### PR DESCRIPTION
**Pull request by javiexin**
Adds core events to includes/acp/acp_profile.php and template events
to adm/style/acp_profile.html to support adding configuration options
to profile fields.

PHPBB3-13911

### CODE REVIEW

File: phpBB/docs/events.md [hunk 1]
```diff
@@ -133,6 +133,13 @@ acp_posting_buttons_before
 * Since: 3.1.0-b4
 * Purpose: Add content before BBCode posting buttons in the ACP
 
+acp_profile_contact_before
+===
+* Locations:
+    + adm/style/acp_profile.html
+* Since: 3.1.5-a4
```
Comment (nickvergessen) [inline phpBB/docs/events.md]: 3.1.5-RC1

File: phpBB/includes/acp/acp_profile.php [hunk 2]
```diff
@@ -369,6 +369,32 @@ function main($id, $mode)
...
+				* @event core.acp_profile_create_edit_init
+				* @var	string	action			create|edit
+				* @var	int		step			Configuration step (1|2|3)
+				* @var	bool	submit			Form has been submitted
+				* @var	bool	save			Configuration should be saved
+				* @var	string	field_type		Type of the field we are dealing with
+				* @var	array	field_row		Array of data about the field
+				* @var	array	exclude			Array of excluded fields by step
+				* @var	array	visibility_ary	Array of fields that are visibility related
+				* @since 3.1.5-a4
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_profile.php]: 3.1.5-RC1

File: phpBB/includes/acp/acp_profile.php [hunk 3]
```diff
@@ -644,6 +670,33 @@ function main($id, $mode)
...
+				* @event core.acp_profile_create_edit_after
+				* @var	string	action			create|edit
+				* @var	int		step			Configuration step (1|2|3)
+				* @var	bool	submit			Form has been submitted
+				* @var	bool	save			Configuration should be saved
+				* @var	string	field_type		Type of the field we are dealing with
+				* @var	array	field_data		Array of data about the field
+				* @var	array	s_hidden_fields	Array of hidden fields in case this needs modification
+				* @var	array	options			Array of options specific to this step
+				* @since 3.1.5-a4
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_profile.php]: 3.1.5-RC1

File: phpBB/includes/acp/acp_profile.php [hunk 4]
```diff
@@ -852,6 +905,25 @@ function save_profile_field(&$cp, $field_type, $action = 'create')
...
+		$field_data = $cp->vars;
+		/**
+		* Event to modify profile field configuration data before saving to database
+		*
+		* @event core.acp_profile_create_edit_before_save
+		* @var	string	action			create|edit
+		* @var	string	field_type		Type of the field we are dealing with
+		* @var	array	field_data		Array of data about the field
+		* @var	array	profile_fields	Array of fields to be sent to the database
+		* @since 3.1.5-a4
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_profile.php]: 3.1.5-RC1

File: phpBB/includes/acp/acp_profile.php [hunk 5]
```diff
@@ -852,6 +905,25 @@ function save_profile_field(&$cp, $field_type, $action = 'create')
 			'field_contact_url'		=> $cp->vars['field_contact_url'],
 		);
 
+		$field_data = $cp->vars;
+		/**
+		* Event to modify profile field configuration data before saving to database
+		*
+		* @event core.acp_profile_create_edit_before_save
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_profile.php]: before should be at the end of the name

**Comment by javiexin** [comment]
Fixed version number and event name.

**Comment by javiexin** [comment]
Will this make it to 3.1.5?  Given the changes requested (@since 3.1.5-RC1) I would have said yes, but it seems this is not included in the current 3.1.5 version, and branch is still open/not merged.
A pity if the @since is the cause for a delay...  Also given that the requested changes were implemented in a couple of hours at most...

**Comment by javiexin** [comment]
Closing this PR as it is against develop-ascareus, reopen against 3.1.x

**Pull request closed without merging**